app/src/main/java/ca/wheresthebus/ui/home/AddFavBottomSheet.kt

```diff
@@ -15,22 +26,38 @@ class AddFavBottomSheet : BottomSheetDialogFragment() {
     // onDestroyView.
     private val binding get() = _binding!!
 
+    private lateinit var stopSuggestionsView: RecyclerView
+    private lateinit var stopSuggestionAdapter: StopSuggestionAdapter
+    private var suggestedStops: ArrayList<BusStop> = arrayListOf()
+
+    private lateinit var mainDBViewModel: MainDBViewModel
+    private lateinit var modelFactory: ModelFactory
+
+    @RequiresApi(Build.VERSION_CODES.O)
     override fun onCreateView(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View? {
+    ): View {
         inflater.inflate(R.layout.bottom_sheet_add_fav, container, false)
-
+        mainDBViewModel = ViewModelProvider(requireActivity()).get(MainDBViewModel::class.java)
+        modelFactory = ModelFactory()
         _binding = BottomSheetAddFavBinding.inflate(inflater, container, false)
         val root: View = binding.root
+        stopSuggestionsView = binding.recyclerViewSuggestions
+        stopSuggestionsView.layoutManager = LinearLayoutManager(context)
+        stopSuggestionAdapter = StopSuggestionAdapter(suggestedStops, mainDBViewModel, modelFactory)
+        stopSuggestionsView.adapter = stopSuggestionAdapter
 
         binding.apply {
             searchViewBus.setupWithSearchBar(searchBarBus)
 
             searchViewBus.editText.setOnEditorActionListener { v, actionId, event ->
                 searchViewBus.hide()
                 searchBarBus.setText(searchViewBus.text)
+                suggestedStops.clear()
+                suggestedStops.addAll(mainDBViewModel.searchByCode(searchViewBus.text.toString()))
+                stopSuggestionAdapter.notifyDataSetChanged()
```

Comment:
(whoops I think this was also my code) Can the results show up as the user inputs them? So instead of having to search for the entire number it shows any stops that contains some of the current input value

Reply:
👍